compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -629,72 +624,55 @@ trait ParallelTesting extends RunnerOrchestration { self =>
-          case Some(i) => JExecutors.newWorkStealingPool(i)
-          case None => JExecutors.newWorkStealingPool()
-        }
-
+      if filteredSources.nonEmpty then
+        val pool = JExecutors.newWorkStealingPool(threadLimit.getOrElse(Runtime.getRuntime.availableProcessors()))
         val timer = new Timer()
         val logProgress = isInteractive && !suppressAllOutput
         val start = System.currentTimeMillis()
-        if (logProgress) {
-          val task = new TimerTask {
-            def run(): Unit = updateProgressMonitor(start)
-          }
-          timer.schedule(task, 100, 200)
-        }
+        if logProgress then
+          timer.schedule(() => updateProgressMonitor(start), 100/*ms*/, 200/*ms*/)
 
-        val eventualResults = filteredSources.map { target =>
+        val eventualResults = for target <- filteredSources yield
           pool.submit(encapsulatedCompilation(target))
-        }
 
         pool.shutdown()
 
-        if (!pool.awaitTermination(20, TimeUnit.MINUTES)) {
+        if !pool.awaitTermination(20, TimeUnit.MINUTES) then
           val remaining = new ListBuffer[TestSource]
-          filteredSources.lazyZip(eventualResults).foreach { (src, res) =>
-            if (!res.isDone)
+          for (src, res) <- filteredSources.lazyZip(eventualResults) do
+            if !res.isDone then
               remaining += src
-          }
 
           pool.shutdownNow()
           System.setOut(realStdout)
           System.setErr(realStderr)
           throw new TimeoutException(s"Compiling targets timed out, remaining targets: ${remaining.mkString(", ")}")
-        }
+        end if
 
-        eventualResults.foreach { x =>
-          try x.get()
-          catch {
-            case ex: Exception =>
-              System.err.println(ex.getMessage)
-              ex.printStackTrace()
-          }
-        }
+        for fut <- eventualResults do
+          try fut.get()
+          catch case ex: Exception =>
+            System.err.println(ex.getMessage)
+            ex.printStackTrace()
 
-        if (logProgress) {
+        if logProgress then
           timer.cancel()
-          val timestamp = (System.currentTimeMillis - start) / 1000
-          realStdout.println(
-            s"[=======================================] completed ($sourceCount/$sourceCount, $failureCount failed, ${timestamp}s)"
-          )
-        }
+          finishProgressMonitor(start)
 
-        if (didFail) {
+        if didFail then
           reportFailed()
           failedTestSources.toSet.foreach(addFailedTest)
           reproduceInstructions.foreach(addReproduceInstruction)
-        }
         else reportPassed()
-      }
       else echo {
         testFilter match
           case _ :: _ => s"""No files matched "${testFilter.mkString(",")}" in test"""
-          case _      => "No tests available under target - erroneous test?"
+          case _ => "No tests available under target - erroneous test?"
```

Comment:
This is a good example of why `case _ =>` is not visually great.

Below `case _ :: _ =>`, it looks like "half" of that pattern. Whereas `case =>` is just whatever.

Reply:
Ironically I'm against this change, but I didn't notice before committing or pushing as I commonly take whitespace changes out. 😄